client/app/bundles/course/users/components/UserProfileCard.tsx

```diff
@@ -0,0 +1,193 @@
+import PropTypes from 'prop-types';
+import { FC } from 'react';
+import {
+  Avatar,
+  Box,
+  Card,
+  CardContent,
+  Grid,
+  Link,
+  Paper,
+  Typography,
+} from '@mui/material';
+import { scroller } from 'react-scroll';
+import { blue, green, orange } from '@mui/material/colors';
+
+interface Props {
+  name: string;
+  imageUrl: string;
+  email: string;
+  role: string;
+  exp?: number;
+  level?: number;
+  experiencePointsRecordsUrl?: string;
+  manageEmailSubscriptionUrl?: string;
+  achievementCount?: number;
+}
+
+const UserProfileCard: FC<Props> = ({
+  name,
+  imageUrl,
+  email,
+  role,
+  exp,
+  level,
+  experiencePointsRecordsUrl,
+  manageEmailSubscriptionUrl,
+  achievementCount,
+}: Props) => {
+  const styles = {
+    courseUserImage: {
+      height: 140,
+      width: 140,
+    },
+    statsContainer: {
+      '& a': {
+        textDecoration: 'none',
+      },
+      '& .user-stats-card': {
+        margin: '4px 4px 4px 0px',
+        padding: '12px',
+        minWidth: '100px',
+        maxWidth: '144px',
+        transition: 'background-color 0.5s, color 0.5s',
+      },
+      '& :nth-child(2n):hover .user-stats-card,& .user-stats-card:nth-child(2n):hover':
+        {
+          backgroundColor: blue[400],
+        },
+      '& :nth-child(3n):hover .user-stats-card,& .user-stats-card:nth-child(3n):hover':
+        {
+          backgroundColor: orange[400],
+        },
+      '& .user-stats-card:hover': {
+        backgroundColor: green[400],
+        color: 'white',
+      },
+    },
+  };
+
+  function UserStatsCard(props): JSX.Element {
+    return (
+      <Paper
+        variant="outlined"
+        className={`user-stats-card ${props.className}`}
+      >
+        <Typography variant="overline">{props.title}</Typography>
+        <Typography variant="h5">{props.value}</Typography>
+      </Paper>
+    );
+  }
+
+  UserStatsCard.propTypes = {
```

Comment:
Why do we have propTypes here? I suppose we are already using TS and thus, we dont need this anymore

Reply:
You can declare a local interface instead for TS